goal_src/jak2/engine/entity/entity-table.gc

```diff
@@ -72,7 +72,7 @@
                           :heap-size #x10000
                           )
                         (new 'static 'entity-info
-                          :ptype (type-ref nav-graph :method-count 47)
+                          :ptype (type-ref nav-graph :method-count 45)
```

Comment:
this looks super suspicious that it changed, but it's actually fine.

For background, in order to insert a reference to a type, the linker has to know the number of methods. This is because the linker needs to get a pointer to a type object, and if it doesn't exist already, it will allocate an "empty" type, and that needs to have the right size.

Here's what happened:

Prior to this PR, the `nav-graph` type didn't exist in `all-types.gc`. The decompiler still has an idea of how many methods the type has because it's contained in the link data. However, in Jak 2, to save space, they store number of methods / 4. So to be safe, the original runtime (and the decompiler) will make room for (stored_value * 4) + 3 methods.

Then, you added `nav-graph` type. The `deftype` flags do have the exact method number, which was only 45.  (45/4) == (47/4), so this change doesn't actually change the output for this file, and everything is ok.

Reply:
Ah good to know the reason, i was similarly incredibly confused/suspicious with this lol